packages/core/src/raa/utils.ts

```diff
@@ -40,14 +40,18 @@ export function cheapestDisplayedRate(hotelOfferEntity: HotelOfferEntity) {
 function isOfferInPriceRange(
   offer: Offer,
   upperBound: number,
-  priceMin: number,
-  priceMax: number
+  priceMin?: number,
+  priceMax?: number
 ) {
-  if (offer.nightlyRate < priceMin) {
+  if (priceMin !== undefined && offer.nightlyRate < priceMin) {
     return false
   }
 
-  if (priceMax < upperBound && offer.nightlyRate > priceMax) {
+  if (
+    priceMax !== undefined &&
+    priceMax < upperBound &&
+    offer.nightlyRate > priceMax
+  ) {
     return false
   }
 
```

Comment:
It took some time to grok this, maybe this is clearer?

```suggestion
return (
  priceMax === undefined ||
  priceMax >= upperBound &&
  offer.nightlyRate <= priceMax
)
```

Reply:
I'm not sure about it, if we do it this way then we also need to include conditions for priceMin here and then it'll become much harder to read.
How I read it now now:
1. If priceMin exists and some other conditions match - return false
2. Then if priceMax exists and some other conditions match - return false
3. Otherwise - return true
4. If I need to dig deeper - I check the specific case.

With the new way I'll read this as:
1. return true or false when I know and have checked all the conditions regardless of the passed params. Personally I found it much harder to read if we have more than 2-3 conditions combined together.

Reply:
I agree splitting the min and max is OK, though for me the negated return value and a default return of `true` is a bit confusing always, and if you miss a condition you end up returning `true` by accident perhaps. 

As the name of the function is nicely descriptive, if I come to check this routine I am probably here for 4 to dive deeper so I think we should optimize for that (I am looking for something like price is within [min, max] but with the current logic it's harder to find that IMO). 

For now implementing the fix is more important, so let's add a expiring TODO for SAPI 1.0.0 to revisit after we have a unit test. Maybe you could specify what the upper bound means in a comment, as that's the main thing confusing me at the moment :)

Reply:
`upperBound` is the highest known historical price (converted from usd to selected currency) for the destination/area. Actually I need to double check if we still need it in SAPI. The algorithm came from daedalus and I don't remember why do we need it for client side filtering.